### PR TITLE
Major refactor of cartpole sample

### PR DESCRIPTION
**Pull request by shnayder**
* updated physics to match cartpole-py sample used in the platform
* updated variable names to match platform Inkling
* updated local rendering to include target pole position
* exposed several additional config vars
* Cleaned up command line params
* Made tests pass again
* Updated README
* Updated and hooked up description json
* Includes Ali's support for connecting to a local exported brain

I've confirmed:

- [x] tests pass
- [x] sim registration info shows up
- [x] local random policy works
- [x] local exported brain works
- [x] rendering works
- [x] training works with platform-sample Inkling
- [x] packaging and running as a managed sim works

### CODE REVIEW

**Comment by akzaidi** [comment]
`pyglet < 1.5.11` appears to [break on MacOS big sur](https://github.com/pyglet/pyglet/issues/274) (which I just updated on Monday):

```bash
Checking /Users/alizaidi-msft/Documents/bonsai/microsoft-bonsai-api/Python/samples/cartpole/logs
Traceback (most recent call last):
  File "main.py", line 481, in <module>
    test_policy(render=args.render, log_iterations=args.log_iterations, policy=random_policy)
  File "main.py", line 270, in test_policy
    sim.episode_step(action)
  File "main.py", line 173, in episode_step
    self.sim_render()
  File "main.py", line 176, in sim_render
    from sim import render
  File "/Users/alizaidi-msft/Documents/bonsai/microsoft-bonsai-api/Python/samples/cartpole/sim/render.py", line 9, in <module>
    from pyglet.gl import (gl, glBegin, glBlendFunc, glClearColor, glColor4f, 
  File "/Users/alizaidi-msft/miniconda3/envs/bonsai-preview/lib/python3.7/site-packages/pyglet/gl/__init__.py", line 95, in <module>
    from pyglet.gl.lib import GLException
  File "/Users/alizaidi-msft/miniconda3/envs/bonsai-preview/lib/python3.7/site-packages/pyglet/gl/lib.py", line 147, in <module>
    from pyglet.gl.lib_agl import link_GL, link_GLU, link_AGL
  File "/Users/alizaidi-msft/miniconda3/envs/bonsai-preview/lib/python3.7/site-packages/pyglet/gl/lib_agl.py", line 43, in <module>
    gl_lib = pyglet.lib.load_library(framework='/System/Library/Frameworks/OpenGL.framework')
  File "/Users/alizaidi-msft/miniconda3/envs/bonsai-preview/lib/python3.7/site-packages/pyglet/lib.py", line 124, in load_library
    return self.load_framework(kwargs['framework'])
  File "/Users/alizaidi-msft/miniconda3/envs/bonsai-preview/lib/python3.7/site-packages/pyglet/lib.py", line 279, in load_framework
    raise ImportError("Can't find framework %s." % path)
```

Upgrading our pyglet in requirements.txt (to say 1.5.15) resolves it though.

File: Python/samples/cartpole/machine_teacher.ink [hunk 1]
```diff
@@ -7,44 +7,78 @@ using Goal
...
             }
         }
     }
-}
\ No newline at end of file
+}
+
+# Special string to hook up the simulator visualizer
+# in the web interface.
+const SimulatorVisualizer = "/cartpoleviz/"
```
Comment (akzaidi) [inline Python/samples/cartpole/machine_teacher.ink:84]: awesome we can keep leveraging the web visualizer again!

**Review by akzaidi — COMMENTED** [review]
LGTM! Just going to run a linter (black) and do a test of the sim-package with an upgraded requirements.txt for pyglet upgrades.

File: Python/samples/cartpole/main.py [hunk 2]
```diff
@@ -224,8 +253,11 @@ def test_random_policy(
         number of iterations to run, by default 10
     """
 
+    policy_name = "random" # TODO
```
Comment (akzaidi) [inline Python/samples/cartpole/main.py:256]: this is left as a TODO. would probably be hacky/weird to convert the `policy` object into a string so you can use it for the logger name, so should we just add an additional argument to the function for now?

**Review by akzaidi — APPROVED** [review]

**Comment by ForrestTrepte** [comment]
This is looking great so far! To complete my review I'd like to try running a couple tests tomorrow, partly for my own edification.

File: Python/samples/cartpole/main.py (hunk 2 above)
Comment (shnayder) [reply Python/samples/cartpole/main.py]: Good catch. Forgot about that.

**Comment by ForrestTrepte** [comment]
All tests worked. I could even use the SimulatorVisualizer while training. Hot!

**Comment by shnayder** [comment]
Ok, I confirmed with Nav offline that he's ok to land this, so going to merge it.

**Review by ForrestTrepte — APPROVED** [review]
Nice work, @shnayder and @akzaidi! All the cleanup and consistency with the default workspace version of cartpole is wonderful.

**Pull request merged**